spring-graphql/src/main/java/org/springframework/graphql/execution/AbstractGraphQlSourceBuilder.java

```diff
@@ -109,7 +117,12 @@ private GraphQLSchema applyTypeVisitors(GraphQLSchema schema) {
 		visitors.add(ContextDataFetcherDecorator.TYPE_VISITOR);
 
 		GraphQLCodeRegistry.Builder codeRegistry = GraphQLCodeRegistry.newCodeRegistry(schema.getCodeRegistry());
-		Map<Class<?>, Object> vars = Collections.singletonMap(GraphQLCodeRegistry.Builder.class, codeRegistry);
+		SubscriptionExceptionResolver subscriptionExceptionResolver = new DelegatingSubscriptionExceptionResolver(
+				subscriptionExceptionResolvers);
+
+		Map<Class<?>, Object> vars = new HashMap<>();
+		vars.put(GraphQLCodeRegistry.Builder.class, codeRegistry);
+		vars.put(SubscriptionExceptionResolver.class, subscriptionExceptionResolver);
```

Comment:
Instead of using a variable, we could pass the `SubscriptionExceptionResolver` (and probably also `GraphQLCodeRegistry`) directly to the `GraphQLTypeVisitor`, which is currently a static variable in `ContextDataFetcherDecorator` but could be a static factory method:
```java
final class ContextDataFetcherDecorator implements DataFetcher<Object> {

    // ...

    static GraphQLTypeVisitor createVisitor(SubscriptionExceptionResolver subscriptionExceptionResolver) {
        // ...
    }
}
```